opm/grid/GraphOfGrid.cpp

```diff
@@ -185,5 +185,30 @@ void GraphOfGrid<Grid>::addWell (const std::set<int>& well, bool checkIntersecti
     }
 }
 
+template<typename Grid>
+void GraphOfGrid<Grid>::addWellBuffer ()
+{
+    // mark all cells that will be added to wells (addding them one
+    // by one would require recursive checks for neighboring wells)
+    std::vector<std::set<int>> buffer(wells.size());
+    int i=0;
+    for (auto& w : wells)
+    {
+        buffer[i].insert(*w.begin()); // intersects with its well
```

Comment:
Coud `w` be empty? Then we need to cater for this.

Reply:
deref'ing begin() on an empty container invokes undefined behavior.

Reply:
begin() can not be dereferenced on an empty container because the for-cycle has zero length.

Reply:
you are talking about ```wells```, we are talking about ```w```.

Reply:
Each well has `size()>1`.
Their function is to keep track of vertices that got contracted and are no longer present in the graph, so each well has a one node (smallest ID) that is still present in the graph and others that got removed.

Reply:
so iow; the answer to the initial question is 'no', not 'yes'. ```w``` cannot be empty.